File: filelocks/read.php
<?php

$filename = "./test.txt";
$filename = "/tmp/test.txt";

$handle = fopen($filename, 'r');

echo "waiting for lock\n";
flock($handle,  LOCK_SH);

echo "locked\n";
echo "read: " . fread($handle, filesize($filename));

sleep(5);

flock($handle, LOCK_UN);
fclose($handle);
echo "unlocked\n";

File: filelocks/read.py
import fcntl
import os
import time

filename = "./test.txt"
filename = "/tmp/test.txt"

with open(filename, 'r') as handle:
  print("waiting for lock")
  fcntl.flock(handle, fcntl.LOCK_SH)

  print("locked")
  print("read: ", handle.read())

  time.sleep(5)

  fcntl.flock(handle, fcntl.LOCK_UN)
  handle.close()
  print("unlocked")
#end_with

